### ISSUE TITLE
Use docker-py to check container configuration in docker states

### ISSUE THREAD

**Issue by terminalmage** [saltstack/salt#39530]
The docker states from were written before support for creating container/host/networking configuration dicts was present in docker-py. This resulted in the need to manually map API arguments to their location in the return data from a container inspection.

We should be able to greatly simplify the logic in the ``docker.running`` state by instead using the passed API arguments to create a container configuration and then match this against the current state of the container.

**Comment by terminalmage** [comment]
This is done aside from some documentation and a blurb in the release notes. I'll have a PR opened for this later today. However, it should not immediately be merged as it includes integration tests for docker which need busybox (to create an image in the setUpClass), and busybox is not in EPEL. I prepared a SRPM on Sunday and sent it to @dmurphy18 to be built, signed and added to our el7 repo. When I open my PR, the tests will be skipped since busybox won't be present, so we'll need to kick off a new run of the test suite once busybox is in our el7 repo.

I've also opened https://github.com/saltstack/salt-jenkins/pull/283 to ensure that docker and busybox are available on the cent7 VMs. This will need to be merged before we kick off a new run of tests.
